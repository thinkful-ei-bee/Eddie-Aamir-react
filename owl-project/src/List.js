import React from 'react';
import Participant from './Participant.js';

function List(props){
  const list = props.participants.map(p => <Participant key={p.id} name={p.name} avatar={p.avatar} inSession={p.inSession} onStage={p.onStage} />);
  return (
    <section className="List">
      {list}
    </section>
  )
}

export default List;