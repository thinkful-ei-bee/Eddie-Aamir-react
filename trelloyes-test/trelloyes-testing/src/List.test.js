import React from 'react';
import ReactDOM from 'react-dom';
import List from './List.js';

describe('<List />', ()=>{
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<List header={'Test Header'} cards={[{title:'Test Title',content:'Test Content'}]}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});
