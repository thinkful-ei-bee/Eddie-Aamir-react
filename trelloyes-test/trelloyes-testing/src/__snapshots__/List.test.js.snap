// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> render the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      Test Header
    </h2>
    <div
      className="List-cards"
    >
      <div
        className="Card"
      >
        <h3>
          Test Title
        </h3>
        <p>
          Test Content
        </p>
      </div>
    </div>
  </header>
</section>
`;
